### PR TITLE
Feature/save players in state

### LINKED ISSUE #14: Save players in state 🍑
This also means having a way of:
- Adding players
- Removing them or declaring that they have forfeited the game
- Updating whom has the turn

### PR DESCRIPTION
**Pull request by amalieboegild**
I have added functionality to add and remove players from the state, and to change turn.

In the main game loop I am updating the turn when I get back a client message. This could for example be saying that Albert just played moves xyz for 20 points. I then update the turn to Albert. But what I want your feedback on is whether this makes sense, since Albert just played? But how should I know who is next after Albert? As far as I can see in the documentation there is no client message saying whose turn it is next, only who just played.. Hmm. Maybe I am overthinking it, but take a look

(Please note that in the main game loop we still need to update board, hand, etc. For now I just pass on the old board and hand)

### CODE REVIEW

File: Scrabble.Bot/Scrabble.fs [hunk 1]
```diff
@@ -145,6 +153,9 @@ module Scrabble =
         //let dict = dictf false // Uncomment if using a trie for your dictionary
         let board = Parser.mkBoard boardP
 
+        let players = Array.create ((int) numPlayers) true
```
Comment (arnenator) [inline Scrabble.Bot/Scrabble.fs]: Maybe this function should have a better name in order to explain what it actually do. Which is create an array of players. So a name could be `createPlayers` or `initializePlayers`.

File: Scrabble.Bot/Scrabble.fs [hunk 2]
```diff
@@ -43,22 +43,24 @@ module State =
...
+        board         : Parser.board
+        dict          : ScrabbleUtil.Dictionary.Dict
+        playerId  : uint32
+        hand          : MultiSet.MultiSet<uint32>
+        players       : array<bool>
+        playerTurn    : uint32
+    }
+
+    let mkState b d pn h pl pt = {board = b; dict = d;  playerId = pn; hand = h; players = pl; playerTurn = pt}
+    let removePlayer st player = st.players[player-1] <- false
```
Comment (arnenator) [inline Scrabble.Bot/Scrabble.fs]: Is it a good idea to mutate the array values?

File: Scrabble.Bot/Scrabble.fs (hunk 1 above)
Comment (amalieboegild) [reply Scrabble.Bot/Scrabble.fs]: fuck off

File: Scrabble.Bot/Scrabble.fs (hunk 2 above)
Comment (amalieboegild) [reply Scrabble.Bot/Scrabble.fs]: I have changed it to lists.

**Review by arnenator — APPROVED** [review]
Looking good

**Pull request merged**